packages/xstate-wallet/src/workflows/application.ts

```diff
@@ -293,11 +296,17 @@ export const applicationWorkflow = (
       const entry = await store.getEntry(context.channelId);
       await messagingService.sendResponse(context.requestId, await convertToChannelResult(entry));
     },
-    spawnObservers: assign<ChannelIdExists>(context => ({
-      ...context,
-      updateObserver: spawn(notifyOnUpdate(context)),
-      requestObserver: spawn(notifyOnChannelRequest(context))
-    })),
+    spawnObservers: assign<ChannelIdExists>((context: WorkflowContext & ChannelIdExists) => {
+      if (!context.requestObserver || !context.updateObserver) {
+        return {
+          ...context,
+          updateObserver: spawn(notifyOnUpdate(context)),
+          requestObserver: spawn(notifyOnChannelRequest(context))
+        };
+      } else {
+        return context;
+      }
```

Comment:
IMO, this would be more clear as
```
    spawnObservers: assign<ChannelIdExists>(context => ({
      ...context,	
      updateObserver: context.updateObserver || spawn(notifyOnUpdate(context)),	
      requestObserver: context.requestObserver || spawn(notifyOnChannelRequest(context))	
    })),	
```